invenio_search_ui/views.py

```diff
@@ -48,91 +48,183 @@ def sorted_options(sort_options):
+            default_sort=None, list_view=True, grid_view=True,
+            pagination_options=[10], aggs={}, default_size=10, **kwargs):
+        """Generate the JSON configuration for Invenio-Search-JS."""
+        config = {
+            "appId": app_id,
+            "initialQueryState": {
+                'hiddenParams': hidden_params,
+            },
+            "searchApi": cls.searchApi(api_list_route, api_mimetype, headers),
+            "sortOptions": cls.sort_options(sort_options, default_sort),
+            "aggs": cls.aggregations(aggs),
+            "layoutOptions": {
+                "listView": list_view,
+                "gridView": grid_view,
+            },
+            "paginationOptions": cls.pagination_options(
+                pagination_options, default_size)
+
+        }
+        config.update(kwargs)
+        return config
+
+    @classmethod
+    def generate_from_rest(
+            cls, config, endpoint_name, hidden_params={}, app_id='search',
+            list_view=True, grid_view=True, pagination_options=[10],
+            default_size=10, **kwargs):
+        """Create JSON config for Invenio-Search-JS with InvenioREST config."""
+        rest_endpoint = config["RECORDS_REST_ENDPOINTS"][endpoint_name]
+        api_list_route = "/api{}".format(rest_endpoint["list_route"])
+        api_mimetype = rest_endpoint["default_media_type"]
+        search_index = rest_endpoint["search_index"]
+        sort_options = config.get("RECORDS_REST_SORT_OPTIONS", {}).get(
+            search_index, {}
+        )
+        default_sort = config.get("RECORDS_REST_DEFAULT_SORT", {}).get(
+            search_index, {}
+        )
+        aggs = (
+            config.get("RECORDS_REST_FACETS", {})
+            .get(search_index, {})
+            .get("aggs", {})
+        )
+        config = {
+            "appId": app_id,
+            "initialQueryState": {
+                'hiddenParams': hidden_params,
+            },
+            "searchApi": cls.searchApi(api_list_route, api_mimetype),
+            "sortOptions": cls.sort_options(sort_options, default_sort),
+            "aggs": cls.aggregations(aggs),
+            "layoutOptions": {
+                "listView": list_view,
+                "gridView": grid_view,
+            },
+            "paginationOptions": cls.pagination_options(
+                pagination_options, default_size)
+
         }
-    }
-    config.update(kwargs)
-    return config
+        config.update(kwargs)
+        return config
+
+
+@blueprint.app_template_global("format_config_from_rest")
+def format_config_from_rest(config, endpoint_name, app_id='search',
+                            hidden_params=None, **kwargs):
+    """Makes Invenio-Search-JS config generation available for Jinja."""
+    return config['SEARCH_UI_SEARCH_CONFIG_GEN'].generate_from_rest(
+        config=config, endpoint_name=endpoint_name, app_id=app_id,
+        hidden_params=hidden_params)
```

Comment:
Based on the config change above this one could become a `bp.app_context_processor` that returns the dictionary with the config objects' `.generate()` classmethod... Let's maybe discuss over chat over specifics of the design.